tests/src/JIT/Directed/arglist/vararg.cs

```diff
@@ -0,0 +1,1280 @@
+            long expectedSum = test_passing_longs(expectedValues.Length, __arglist(expectedValues[0], expectedValues[1], expectedValues[2], expectedValues[3]));
+
+            long sum = 0;
+            for (int i = 0; i < expectedValues.Length; ++i)
+            {
+                sum += expectedValues[i];
+            }
+
+            return sum == expectedSum;
+        }
+
+        /// <summary>
+        /// Given an input set create an arglist to pass to a vararg callee.
+        /// 
+        /// The callee will simply loop over the arguments, compute the sum
+        /// then return the value.
+        /// 
+        /// Do a quick check on the value returned, and return whether they 
+        /// are equal.
+        /// 
+        /// </summary>
+        /// <param name="expectedValues"></param>
+        /// <returns>bool</returns>
+        static bool TestPassingFloats(float[] expectedValues)
+        {
+            Debug.Assert(expectedValues.Length == 4);
+            float expectedSum = test_passing_floats(expectedValues.Length, __arglist(expectedValues[0], expectedValues[1], expectedValues[2], expectedValues[3]));
+
+            float sum = 0;
+            for (int i = 0; i < expectedValues.Length; ++i)
+            {
+                sum += expectedValues[i];
+            }
+
+            return sum == expectedSum;
+        }
+
+        /// <summary>
+        /// Given an input set create an arglist to pass to a vararg callee.
+        /// 
+        /// The callee will simply loop over the arguments, compute the sum
+        /// then return the value.
+        /// 
+        /// Do a quick check on the value returned, and return whether they 
+        /// are equal.
+        /// 
+        /// </summary>
+        /// <param name="expectedValues"></param>
+        /// <returns>bool</returns>
+        static bool TestPassingDoubles(double[] expectedValues)
+        {
+            Debug.Assert(expectedValues.Length == 4);
+            double expectedSum = test_passing_doubles(expectedValues.Length, __arglist(expectedValues[0], expectedValues[1], expectedValues[2], expectedValues[3]));
+
+            double sum = 0;
+            for (int i = 0; i < expectedValues.Length; ++i)
+            {
+                sum += expectedValues[i];
+            }
+
+            return sum == expectedSum;
+        }
+
+        /// <summary>
+        /// Given an input set create an arglist to pass to a vararg callee.
+        /// 
+        /// The callee will simply loop over the arguments, compute the sum
+        /// then return the value.
+        /// 
+        /// Do a quick check on the value returned, and return whether they 
+        /// are equal.
+        /// 
+        /// </summary>
+        /// <param name="expectedValues"></param>
+        /// <returns>bool</returns>
+        static bool TestPassingEmptyInts(int[] expectedValues)
+        {
+            int expectedSum = test_passing_ints(expectedValues.Length, __arglist());
```

Comment:
What about:
```
Debug.Assert(expectedValues.Length == 0);
```
and no "sum" loop?

In fact, why even take an `expectedValues` arg?